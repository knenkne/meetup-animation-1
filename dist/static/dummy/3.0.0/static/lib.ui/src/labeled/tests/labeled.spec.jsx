import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import { Labeled } from '..'

describe('Labeled', () => {
    it('is available', () => {
        expect(Labeled).toBeDefined()
    })

    it('renders children', () => {
        const wrapper = shallow(
            <Labeled>
                <span>{'foo bar baz'}</span>
            </Labeled>
        )

        expect(wrapper.contains(<span>{'foo bar baz'}</span>)).toBeTruthy()
    })

    it('renders title with headline class', () => {
        const tree = renderer
            .create(
                <Labeled title="Hello">
                    <span>{'foo bar baz'}</span>
                </Labeled>
            )
            .toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('renders tooltip with headline class', () => {
        const tree = renderer
            .create(
                <Labeled tooltip={{ contents: 'Подсказка' }}>
                    <span>{'foo bar baz'}</span>
                </Labeled>
            )
            .toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('renders description without headline class', () => {
        const tree = renderer
            .create(
                <Labeled description="Hello">
                    <span>{'foo bar baz'}</span>
                </Labeled>
            )
            .toJSON()

        expect(tree).toMatchSnapshot()
    })


})
