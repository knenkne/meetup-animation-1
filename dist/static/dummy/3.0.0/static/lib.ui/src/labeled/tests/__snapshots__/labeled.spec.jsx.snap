// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labeled renders description without headline class 1`] = `
<div
  className="labeled undefined"
>
  <div>
    <span>
      foo bar baz
    </span>
  </div>
  <div
    className="description"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Hello</p>
",
      }
    }
    data-unit="markdown:full"
  />
</div>
`;

exports[`Labeled renders title with headline class 1`] = `
<div
  className="labeled undefined"
>
  <div
    className="headline"
  >
    <span
      className="caption title"
    >
      Hello
    </span>
  </div>
  <div>
    <span>
      foo bar baz
    </span>
  </div>
</div>
`;

exports[`Labeled renders tooltip with headline class 1`] = `
<div
  className="labeled undefined"
>
  <div
    className="headline"
  >
    <div
      className="hint"
    >
      <div
        className="tooltip"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="tool"
          data-unit="tooltip:title"
          onClickCapture={[Function]}
        >
          <button
            aria-controls="tooltip-1"
            aria-describedby="tooltip-1"
            className="info sm"
            data-unit="button:info:link"
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="icon icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"36\\" height=\\"36\\" viewBox=\\"0 0 36 36\\"><path d=\\"M18 28c5.523 0 10-4.477 10-10S23.523 8 18 8 8 12.477 8 18s4.477 10 10 10zm0 2c-6.627 0-12-5.373-12-12S11.373 6 18 6s12 5.373 12 12-5.373 12-12 12z\\"/><path d=\\"M17 17.003A.999.999 0 0118 16c.552 0 1 .438 1 1.003v4.994A.999.999 0 0118 23c-.552 0-1-.438-1-1.003v-4.994zM17 12h2v2h-2v-2z\\"/></svg>",
                }
              }
              data-unit="icon"
              onClick={[Function]}
              onError={[Function]}
            />
            
          </button>
        </div>
        <div
          aria-live="assertive"
          className="tip top left info"
          id="tooltip-1"
          role="tooltip"
        >
          <div
            className="activeZone"
          >
            <div
              className="contents"
            >
              <div
                className="container sm"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>Подсказка</p>
",
                  }
                }
                data-unit="markdown:full"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <span>
      foo bar baz
    </span>
  </div>
</div>
`;
